openmls/src/treesync/node/leaf_node.rs

```diff
@@ -1,20 +1,69 @@
 //! This module contains the [`LeafNode`] struct and its implementation.
+use openmls_traits::{crypto::OpenMlsCrypto, types::CryptoError};
 use serde::{Deserialize, Serialize};
 
 use crate::{
-    ciphersuite::{HpkePrivateKey, HpkePublicKey},
-    key_packages::{KeyPackage, KeyPackageBundle},
+    ciphersuite::{hash_ref::KeyPackageRef, HpkePrivateKey, HpkePublicKey},
+    key_packages::{KeyPackage, KeyPackageBundle, KeyPackageError},
 };
 
 /// This struct implements the MLS leaf node and contains a [`KeyPackage`] and
 /// potentially a corresponding [`HpkePrivateKey`].
-#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct LeafNode {
+    key_package_ref: Option<KeyPackageRef>,
```

Comment:
We don't always need the key package reference and can easily compute it from the `KeyPackage`. So I don't think it makes sense to have it in there. Here it is an identifier for the leaf such that it would be annoying having to recompute it every time.
Unless you have strong feelings about it I'd keep it here.